File: fixtures/python/active_debug_code/positive.py
# Fixture: MAINT011-active-debug-code — Python positive cases
# Each call below should produce a finding.

def process_data(data):
    print(data)          # should be flagged
    return data


def debug_session():
    breakpoint()         # should be flagged


def inspect_value(obj):
    import pdb
    pdb.set_trace()      # should be flagged

File: fixtures/js/active_debug_code/positive.ts
// Fixture: MAINT011-active-debug-code — JS/TS positive cases
// Each construct below should produce a finding.

function fetchData(url: string): Promise<Response> {
    debugger; // should be flagged (Severity::Medium)
    console.log("fetching", url); // should be flagged (Severity::Low)
    return fetch(url);
}

function processResult(data: unknown): void {
    console.debug("processing:", data); // should be flagged (Severity::Low)
    console.trace("stack trace"); // should be flagged (Severity::Low)
}
